packages/dds/tree/src/test/cursorTestSuite.ts

```diff
@@ -546,6 +564,19 @@ function checkTraversal(cursor: ITreeCursor) {
         // skipPendingFields should have no effect since not pending
         assert(cursor.skipPendingFields());
         assert.equal(cursor.getFieldKey(), key);
+
+        // Recursively validate.
+        actualChildNodesTraversed = 0;
+        for (let inNode = cursor.firstNode(); inNode; inNode = cursor.nextNode()) {
```

Comment:
What's the purpose of testing this inside a separate loop?

Reply:
As noted in the comment above, it's easier to debug non-recursive validation, so we do the non-recursive validation first, making sure as much of the cursor is working as practical before we do the recursive check: if basic navigation is broken, the recursive check can produce really confusing errors so I prefer more specific errors I can know were not caused by the recursion and ca easily step through with the debugger.

This means validating both iterating and indexing done on the above loops before recursing.